pkg/apiserver/registry/security/workloadidentity/storage/jwt.go

```diff
@@ -0,0 +1,299 @@
+	UID       string `json:"uid,omitempty"`
+}
+
+type garden struct {
+	ID string `json:"id,omitempty"`
+}
+
+// issueToken generates the JSON Web Token based on the provided configurations.
+func (r *TokenRequestREST) issueToken(tokenRequest *securityapi.TokenRequest, workloadIdentity *securityapi.WorkloadIdentity) (string, *time.Time, error) {
+	signer, err := getSigner(r.signingKey)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to get signer: %w", err)
+	}
+
+	duration := tokenRequest.Spec.ExpirationSeconds
+	if duration < r.minDuration {
+		duration = r.minDuration
+	} else if duration > r.maxDuration {
+		duration = r.maxDuration
+	}
+
+	var (
+		iat = time.Now()
+		exp = iat.Add(time.Second * time.Duration(duration))
+	)
+
+	shoot, seed, project, err := r.resolveContextObject(tokenRequest.Spec.ContextObject)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to resolve context object: %w", err)
+	}
+
+	standardClaims, gardenerClaims := r.getClaims(iat, exp, workloadIdentity, shoot, seed, project)
+	token, err := jwt.Signed(signer).Claims(gardenerClaims).Claims(standardClaims).Serialize()
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to issue JSON Web token: %w", err)
+	}
+
+	return token, &exp, nil
+}
+
+func (r *TokenRequestREST) getClaims(iat, exp time.Time, workloadIdentity *securityapi.WorkloadIdentity, shoot, seed, project metav1.Object) (*jwt.Claims, *gardenerClaims) {
+	standardClaims := &jwt.Claims{
+		Issuer:    r.issuer,
+		Subject:   workloadIdentity.Status.Sub,
+		Audience:  workloadIdentity.Spec.Audiences,
+		IssuedAt:  jwt.NewNumericDate(iat),
+		Expiry:    jwt.NewNumericDate(exp),
+		NotBefore: jwt.NewNumericDate(iat),
+	}
+
+	gardenerClaims := &gardenerClaims{
+		Gardener: gardener{
+			WorkloadIdentity: ref{
+				Name:      workloadIdentity.Name,
+				Namespace: workloadIdentity.Namespace,
+				UID:       string(workloadIdentity.UID),
+			},
+			Garden: &garden{
+				ID: r.clusterIdentity,
+			},
+		},
+	}
+
+	if shoot != nil {
+		gardenerClaims.Gardener.Shoot = &ref{
+			Name:      shoot.GetName(),
+			Namespace: shoot.GetNamespace(),
+			UID:       string(shoot.GetUID()),
+		}
+	}
+
+	if seed != nil {
+		gardenerClaims.Gardener.Seed = &ref{
+			Name: seed.GetName(),
+			UID:  string(seed.GetUID()),
+		}
+	}
+
+	if project != nil {
+		gardenerClaims.Gardener.Project = &ref{
+			Name: project.GetName(),
+			UID:  string(project.GetUID()),
+		}
+	}
+
+	return standardClaims, gardenerClaims
+}
+
+func (r *TokenRequestREST) resolveContextObject(ctxObj *securityapi.ContextObject) (metav1.Object, metav1.Object, metav1.Object, error) {
+	if ctxObj == nil {
+		return nil, nil, nil, nil
+	}
+
+	var (
+		shoot   *gardencorev1beta1.Shoot
+		seed    *gardencorev1beta1.Seed
+		project *gardencorev1beta1.Project
+
+		shootMeta, seedMeta, projectMeta metav1.Object
+		err                              error
+		coreInformers                    = r.coreInformerFactory.Core().V1beta1()
+	)
```

Comment:
Can we move this outside of `var` and above the `gvk` variable using the `:=` operation

Reply:
I could not find any recommendations against this, therefore I would prefer to keep declaration/initialization of multiple variables in a `var` block.